test/instrumentationsTests/http.ts

```diff
@@ -0,0 +1,113 @@
+import { InstrumentationTest } from './InstrumentationTest';
+import { callContainer } from '../helpers/helpers';
+import {ChildProcess} from "child_process";
+
+class HttpInstrumentationTest implements InstrumentationTest {
+  isChildProcessReadyPredicate(data: any, nodeChildApp: ChildProcess, resolve, reject): void {
+    const dataStr = data.toString();
+    if(dataStr.includes("Server is running on http://localhost:8000")){
+      resolve();
+    }
+  }
+
+  onChildProcessReady(): Promise<void> {
+    return callContainer(8000, 'test', 'get');
+  }
+
+  spansReadyCondition(lines: string[], resolve): void {
+    console.log(`Checking [${lines.length}] lines:`);
+    lines.forEach((l) => console.log(l.substring(0, 50)));
+    if (
+      lines.length === 2 &&
+      lines[0].startsWith('{"traceId"') &&
+      lines[1].startsWith('{"traceId"')
+    ) {
+      console.log('Spans are ready!');
+      resolve(lines);
+    }
+  }
+
+  runTests(spans: any[]): void {
+    expect(spans).toHaveLength(2);
+    const internalSpan = spans.find((span) => span.kind === 1);
+    const clientSpan = spans.find((span) => span.kind === 2);
+    expect(internalSpan).toMatchObject({
+      traceId: expect.any(String),
+      id: expect.any(String),
+      timestamp: expect.any(Number),
+      duration: expect.any(Number),
+      name: 'HTTP GET',
+      kind: 1,
+      resource: {
+        attributes: {
+          "service.name": "http-js",
+          "telemetry.sdk.language": "nodejs",
+          "telemetry.sdk.name": "opentelemetry",
+          "telemetry.sdk.version": "1.1.1",
+          "framework": "node",
+          'process.environ': expect.stringMatching(/\{.*\}/),
```

Comment:
It is worth trying to JSON.parse it and see if it is an object. The test like this won't go red on malformed JSON.

Reply:
https://github.com/duailibe/jest-json this looks nice btw

Reply:
again the value is a string not a sjon so we would need to first JSON.parse it